lib/foreman/renderer/scope/macros/base.rb

```diff
@@ -104,14 +104,19 @@ def pxe_kernel_options
           apipie :method, 'Generates a shell command to store the given text into a file' do
             desc "This is useful if some multiline string needs to be saved somewhere on the hard disk. This
               is typically used in provisioning or job templates, e.g. when puppet configuration file is
-              generated based on host configuration and stored for puppet agent."
+              generated based on host configuration and stored for puppet agent. The content must end with
+              a line end, if not an extra trailing line end is appended automatically."
             required :filename, String, desc: 'the file path to store the content to'
             required :content, String, desc: 'content to be stored'
             returns String, desc: 'String representing the shell command'
             example "save_to_file(\"hello\\nworld\\n\", '/etc/motd') # => 'cat << EOF > /etc/motd\\nhello\\nworld\\nEOF'"
           end
           def save_to_file(filename, content)
-            "cat << EOF > #{filename}\n#{content}EOF"
+            if content && content[-1] == "\n"
```

Comment:
not crucial but i think `ends_with?` might be a bit clearer:
```suggestion
            if content&.ends_with?("\n")
```

Reply:
OTH I'd say:
```suggestion
            if !content || content.ends_with?("\n")
```
This way, we wouldn't endup with double line on missing content